server/jetstream_api.go

```diff
@@ -262,7 +262,22 @@ const JSMaxNameLen = 256
 // TODO(dlc) - Move to more generic location.
 type ApiError struct {
 	Code        int    `json:"code"`
+	Kind        string `json:"kind,omitempty"`
 	Description string `json:"description,omitempty"`
+	URL         string `json:"-"`
```

Comment:
Why are these never serialized and sent to clients?

Reply:
These are there mainly to assist tooling - I will add some `nats explain` or something that read these and generate documents, links to further information etc.

Doesnt seem appropraite to put in API responses - its a documentation link.

Reply:
Should these be separate aggregate struct then that extends?

Reply:
Sure, but whats the benefit? I don't see how this is better or worse then?

Reply:
Feels odd to have public members on a struct used to send JSON responses that includes things that we do not send.. But not super important. I would do aggregate (has-a) or not make them public, but again not that critical.

Reply:
They are public so that other tools - like the NATS CLI can get access to them via the Go API rather than have to maintain some additional structures etc.

So for that use case I think this is exactly where they should be, even if I added them in another struct that extend, to allow the CLI to access that extending struct I would have to export that struct key and tell JSON to ignore it. Which feels about exactly the same as this.

Reply:
Not following that part, so you are using the NATS server source code directly in the NATS cli or do you mean the Go client? Which is of course very different and will have its own version of this.

Reply:
Only the CLI

Definitely not any clients - they only need to know the ErrCode which is in the JSON.

(something like)
```
$ nats explain 10039
Message:  JetStream not enabled for account
Help: https://docs.nats.io/jetstream

This error indicates that JetStream is not enabled either at a global level or at global and account level
```

This help url and additional short help comes out of these structs directly

Reply:
However, to be exactly clear about why I did certain things, these errors are all in a JSON files in this repository. The build process of the CLI can pull in that json file and generate the richer structures there so this is all out of the nats server go code but we still end up with one place to maintain the data.

So I can completely remove these fields from the structs here and it would be ok(ish) though more work in the CLI.  But the JSON file is there so that any other language that wish to gain access to this stuff can do so

Reply:
How do we plan on dealing with server / cli version mismatch in this case? Should the json be something you can get from the server's system account maybe?

Reply:
We could for sure do that, we do have quite a bit of version coupling between CLI and Server atm so don't consider it the end of the world.

It's probably something we can improve around the time we do schema registeries - so this is some data that can also provide?

Reply:
Might be worth discussing.